serving/src/main/java/ai/djl/serving/http/KServeRequestHandler.java

```diff
@@ -160,6 +185,199 @@ private void handleKServeDescribeModel(ChannelHandlerContext ctx, String[] segme
         NettyUtils.sendJsonResponse(ctx, response);
     }
 
+    private void handleKServeDescribeInfer(
+            ChannelHandlerContext ctx,
+            FullHttpRequest req,
+            QueryStringDecoder decoder,
+            String[] segments) {
+        String modelName = segments[3];
+        String modelVersion = null;
+        if (segments.length > 5) {
+            modelVersion = segments[5];
+        }
+        Input inferenceRequest = requestParser.parseRequest(req, decoder);
+        infer(ctx, inferenceRequest, modelName, modelVersion);
+    }
+
+    private ArrayList<RequestInput> getInputListFromString(String requestString) {
+        GsonBuilder builder = new GsonBuilder();
+        builder.setPrettyPrinting();
+        Gson gson = builder.create();
+        return gson.fromJson(requestString, new TypeToken<ArrayList<RequestInput>>() {}.getType());
+    }
+
+    private ArrayList<RequestOutput> getOutputListFromString(String requestString) {
+        GsonBuilder builder = new GsonBuilder();
+        builder.setPrettyPrinting();
+        Gson gson = builder.create();
+        ArrayList<RequestOutput> arrayList =
+                gson.fromJson(
+                        requestString, new TypeToken<ArrayList<RequestOutput>>() {}.getType());
+        return arrayList;
+    }
```

Comment:
Instead of parsing it as a string and then converting to RequestInput/RequestOutput, you can directly parse the bytes to the respective class. And if the user given input/output format is wrong, throw the exception.